### PR TITLE
Initial AWS CCM deployment using upstream image

### PR DESCRIPTION
**Pull request by Danil-Grigorev**
Requirements to work:
----
Needs to additionally be merged:
- [ ] https://github.com/openshift/machine-config-operator/pull/2401 - fix for AWS hostname for successful migration
- [ ] https://github.com/openshift/machine-config-operator/pull/2386 - Kubelet FG switch to external providers
- [ ] https://github.com/openshift/library-go/pull/895 (+ revendor in KCMO and KAPIO) - KCM and KAPI FG switch to external providers
- [x] https://github.com/openshift/cluster-cloud-controller-manager-operator/pull/26 - Include in CCCMO in release

### CODE REVIEW

**Comment by Danil-Grigorev** [comment]
/hold cancel

This implementation works in my manual testing. Unit tests are already in place, the PR is ready for review.

File: manifests/0000_26_cloud-controller-manager-operator_03_rbac_provider.yaml [hunk 1]
```diff
@@ -8,57 +8,6 @@ metadata:
...
-    include.release.openshift.io/single-node-developer: "true"
-roleRef:
-  kind: Role
-  name: cloud-controller-manager
-  apiGroup: rbac.authorization.k8s.io
-subjects:
-  - kind: ServiceAccount
-    namespace: openshift-cloud-controller-manager
-    name: cloud-controller-manager
-
```
Comment (Danil-Grigorev) [inline manifests/0000_26_cloud-controller-manager-operator_03_rbac_provider.yaml:61]: AWS covers these RBAC changes with upaded cluster role [here](https://github.com/openshift/cluster-cloud-controller-manager-operator/pull/15/files#diff-0856a88d16f3aa369e7f5d7fc7d1e356008fc796a5800ad4da0e304113f6af2cR60). I consider removing this for now, as for example Azure and vSphere both are using same clusterRole, which will work for OpenStack too. We will investigate and possible reduce the scope of it later.

**Review by Fedosin — CHANGES_REQUESTED** [review]
What about adding unit tests similar to what we have with OpenStack: https://github.com/openshift/cluster-cloud-controller-manager-operator/blob/master/pkg/cloud/openstack/openstack_test.go

**Comment by Danil-Grigorev** [comment]
@Fedosin It is already implemented and covered in https://github.com/openshift/cluster-cloud-controller-manager-operator/blob/master/controllers/clusteroperator_controller_test.go#L271-L336

File: manifests/0000_26_cloud-controller-manager-operator_03_rbac_provider.yaml (hunk 1 above)
Comment (mdbooth) [reply manifests/0000_26_cloud-controller-manager-operator_03_rbac_provider.yaml:61]: Ok, sure. As I mentioned elsewhere I think we could potentially eliminate these entirely later.

File: pkg/cloud/aws.go [hunk 2]
```diff
@@ -1,9 +1,44 @@
...
+//go:embed aws/assets/*
+var f embed.FS
+
 func GetAWSResources() []client.Object {
-	return []client.Object{}
+	resources := []struct {
+		obj   client.Object
+		asset string
+	}{
+		{&appsv1.Deployment{}, "aws/assets/deployment.yaml"},
```
Comment (elmiko) [inline pkg/cloud/aws.go:22]: is the idea here that more resources will be added later?

File: pkg/cloud/aws.go [hunk 3]
```diff
@@ -1,9 +1,44 @@
...
+	}
+	ret := make([]client.Object, 0, len(resources))
+
+	for i := 0; i < len(resources); i++ {
```
Comment (elmiko) [inline pkg/cloud/aws.go:26]: why not use `for ... range` here?
```suggestion
	for _, resource := range resources {
```

**Review by elmiko — COMMENTED** [review]
the code mostly makes sense, i have a few questions inline. also, is the main goal here for the code to dynamically load these manifests at run time?

File: pkg/cloud/aws.go (hunk 2 above)
Comment (Danil-Grigorev) [reply pkg/cloud/aws.go:22]: Correct, it is build with intention that it might need to deploy some SA or another Deployment potentially in the future.

**Review by elmiko — APPROVED** [review]
thanks Danil!
/approve

**Comment by elmiko** [comment]
/hold
didn't realize it added both labels. just putting a hold so others can review

**Comment by Danil-Grigorev** [comment]
/lgtm cancel

We need to fix the CI to prevent double label addition @Fedosin 
Let's wait for tests to pass again.

**Comment by elmiko** [comment]
> Can we make this look more similar to the openstack one? Keeping it in its own package in case it expands later?

+1, i think that makes good sense

**Comment by Danil-Grigorev** [comment]
@JoelSpeed OpenStack manifests are located incorrectly. They should belong to higher level, so cloud provider switch case selection won't have to import N packages per provider just to select correct collection of manifests: see https://github.com/openshift/cluster-cloud-controller-manager-operator/blob/master/controllers/clusteroperator_controller.go#L25-L26

**Pull request merged**